python/lsst/meas/algorithms/brightStarStamps.py

```diff
@@ -0,0 +1,274 @@
+    innerRadius : `int`, optional
+        Inner radius value, in pixels. This and ``outerRadius`` define the
+        annulus used to compute the ``"annularFlux"`` values within each
+        ``starStamp``. Must be provided if ``"INNER_RADIUS"`` and
+        ``"OUTER_RADIUS"`` are not present in ``metadata``.
+    outerRadius : `int`, optional
+        Outer radius value, in pixels. This and ``innerRadius`` define the
+        annulus used to compute the ``"annularFlux"`` values within each
+        ``starStamp``. Must be provided if ``"INNER_RADIUS"`` and
+        ``"OUTER_RADIUS"`` are not present in ``metadata``.
+    metadata : `lsst.daf.base.PropertyList`, optional
+        Metadata associated with the bright stars.
+
+    Raises
+    ------
+    ValueError
+        Raised if one of the star stamps provided does not contain the
+        required keys.
+    AttributeError
+        Raised if the definition of the annulus used to compute each star's
+        normalization factor are not provided, that is, if ``"INNER_RADIUS"``
+        and ``"OUTER_RADIUS"`` are not present in ``metadata`` _and_
+        ``innerRadius`` and ``outerRadius`` are not provided.
+
+    Notes
+    -----
+    A (gen2) butler can be used to read only a part of the stamps,
+    specified by a bbox:
+
+    >>> starSubregions = butler.get("brightStarStamps_sub", dataId, bbox=bbox)
+    """
+
+    def __init__(self, starStamps, innerRadius=None, outerRadius=None,
+                 metadata=None,):
+        for item in starStamps:
+            if not isinstance(item, BrightStarStamp):
+                raise ValueError(f"Can only add instances of BrightStarStamp, got {type(item)}")
+        self._starStamps = starStamps
+        self._metadata = PropertySet() if metadata is None else metadata
+        # Add inner and outer radii to metadata
+        self._checkRadius(innerRadius, "INNER_RADIUS")
+        self._innerRadius = innerRadius
+        self._checkRadius(outerRadius, "OUTER_RADIUS")
+        self._outerRadius = outerRadius
+
+    def __len__(self):
+        return len(self._starStamps)
+
+    def __getitem__(self, index):
+        return self._starStamps[index]
+
+    def append(self, item):
+        """Add an additional bright star stamp.
+
+        Parameters
+        ----------
+        item : `dict`
+            Mapping containing keys ``"starStamp"``, ``"gaiaGMag"``,
+            ``"gaiaId"``, and `"annularFlux"`.
+        """
+        if not isinstance(item, BrightStarStamp):
+            raise ValueError(f"Can only add instances of BrightStarStamp, got {type(item)}.")
+        self._starStamps.append(item)
+        return None
+
+    def extend(self, bss):
+        """Extend BrightStarStamps instance by appending elements from another
+        instance.
+
+        Parameters
+        ----------
+        bss : `BrightStarStamps`
+            Other instance to concatenate.
+        """
+        self._checkRadius(bss._innerRadius, "INNER_RADIUS")
```

Comment:
Though in the case of this class it is rather limited in use, it is best to not use magic strings if it can be avoided. Magic strings are susceptible to typos by someone at some future point, which are impossible to lint, which leads to an error not being found until some future runtime error.

If you can enumerate all the possibilities (such as this case) it is better to use pythons Enum class like follows

```python
from enum import Enum, auto

class RadiiEnum(Enum):
    INNER_RADIUS = auto()  # auto because we dont actually care what the enumeration number is
    OUTER_RADIUS = auto()
```
This allows you to use and check like follows
```python
self._checkRadius(bss._outerRadius, RadiiEnum.INNER_RADIUS)
....
    def _checkRadius(self, radiusValue, metadataEnum):
        if metadataEnum is RadiiEnum.INNER_RADIUS:
            ....
        else:
            ....
```
Linters will then be able to verify everything is correct every time it is used. However we can do one better for your case:
```python
class RadiiEnum(Enum):
    INNER_RADIUS = auto()
    OUTER_RADIUS = auto()

    def __str__(self):
        return self.name #  name is the name of what ever variant the value happened to be
```
This will turn the enum into a string with the correct name when you use it as such:
```python
    def _checkRadius(self, radiusValue, metadataEnum):
        metadataName = str(metadataEnum)
       #  The rest of your function can now remain the same
```
This way users can just use the enum values which will be checked by the linter, editors can auto complete, and it is self documenting in that all the possible options are enumerated (you know that their isnt some other metadata field you could check). The only failure can then be if you spell your enum name wrong, and that can be checked by a simple unit test.